worker/crlWorker/crlWorker.go

```diff
@@ -0,0 +1,200 @@
+	result := worker.Result{
+		WorkerName: workerName,
+	}
+	crlRes := Result{}
+
+	// Grab first CRL response we can from our certificate
+	crlResponses, err := getCRLResponses(in.Certificate)
+	if err != nil {
+		resChan <- worker.Result{
+			Success:    false,
+			WorkerName: workerName,
+			Errors:     []string{fmt.Sprintf("error getting CRL response, err=%v", err)},
+			Result:     nil,
+		}
+		return
+	}
+	if len(crlResponses) == 0 {
+		resChan <- worker.Result{
+			Success:    false,
+			WorkerName: workerName,
+			Errors:     []string{fmt.Sprintf("no CRL responses received, err=%v", err)},
+			Result:     nil,
+		}
+		return
+	}
+
+	// TODO(adam): store all CRL responses
+	// TODO(adam): Process all responses? Or just one that's signed?
+
+	certList, err := x509.ParseCRL(crlResponses[0])
+	if err != nil {
+		resChan <- worker.Result{
+			Success:    false,
+			WorkerName: workerName,
+			Errors:     []string{fmt.Sprintf("error reading CRL response for %s", in.Certificate.ID)},
+			Result:     nil,
+		}
+		return
+	}
+
+	// Verify the CRL
+	if err = verifyCRL(certList, in.Certificate, in.CertificateChain); err != nil {
+		resChan <- worker.Result{
+			Success:    false,
+			WorkerName: workerName,
+			Errors:     []string{fmt.Sprintf("error verifying CRL, err=%v", err)},
+			Result:     nil,
+		}
+		return
+	}
+
+	x509Cert, err := in.Certificate.ToX509()
+	if err != nil {
+		resChan <- worker.Result{
+			Success:    false,
+			WorkerName: workerName,
+			Errors:     []string{fmt.Sprintf("error converting Certificate %s to x509.Certificate, err=%v", in.Certificate.ID, err)},
+			Result:     nil,
+		}
+		return
+	}
+
+	// Check if our certificate is in the revoked list
+	revoked := certList.TBSCertList.RevokedCertificates
+	for i := range revoked {
+		if x509Cert.SerialNumber.Cmp(revoked[i].SerialNumber) == 0 {
+			// certificate is in revoked list, serials match
+			crlRes.RevocationTime = certList.TBSCertList.ThisUpdate
+			crlRes.Revoked = true
+
+			if err := in.DBHandle.UpdateCertMarkAsRevoked(in.Certificate.ID); err != nil {
+				resChan <- worker.Result{
+					Success:    false,
+					WorkerName: workerName,
+					Errors:     []string{fmt.Sprintf("error update Certificate %s revoked_via_crl, err=%v", in.Certificate.ID, err)},
+					Result:     nil,
+				}
+				return
+			}
+			break
+		}
+	}
+
+	// Marshal the response
+	bs, err := json.Marshal(&crlRes)
+	if err != nil {
+		result.Success = false
+		result.Errors = append(result.Errors, err.Error())
+	} else {
+		result.Success = true
+		result.Result = bs
+	}
+	resChan <- result
+}
+
+// Grab the first CRL response and return it in raw bytes
+func getCRLResponses(cert certificate.Certificate) ([][]byte, error) {
+	var wg sync.WaitGroup
+	var out [][]byte
+
+	crlPoints := cert.X509v3Extensions.CRLDistributionPoints
+	wg.Add(len(crlPoints))
+	for i := range crlPoints {
+		go func(point string, wg *sync.WaitGroup) {
+			defer wg.Done()
+
+			resp, err := httpClient.Get(point)
```

Comment:
how about using an http cache like https://github.com/gregjones/httpcache ?
note that I've never used it, so no idea if it'd work